### PR TITLE
Update Go version and dependencies

### PR DESCRIPTION
**Pull request by johejo**
## Description

Update Go version and dependencies.
Fixes some issue reported by new version of golangci-lint.
Only TLS 1.2 and TLS 1.3 are supported for HTTPS.

## Motivation and Context

It's good time to update Go.

## How Has This Been Tested?

No new features are included.
Fixed some broken tests.

Updating some dependencies broke tests.
- mitchellh/mapstructure: locked the same version with viper
- vmihailenco/msgpack/v4: v4.3.12 brokes some tests so locked into v4.3.11

## Checklist:

- [x] My change requires a change to the documentation or CHANGELOG.
- [x] I have updated the documentation/CHANGELOG accordingly.
- [x] I have created a feature (non-master) branch for my PR.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -10,7 +10,7 @@ require (
 	github.com/coreos/go-oidc v2.2.1+incompatible
 	github.com/dgrijalva/jwt-go v3.2.0+incompatible
 	github.com/fsnotify/fsnotify v1.4.9
-	github.com/go-redis/redis/v7 v7.4.0
+	github.com/go-redis/redis/v8 v8.0.0
```
Comment (NickMeves) [inline go.mod:13]: Is this change required in the update to the new Go version as a dependency somewhere?

Otherwise can we move this to its own PR to split up the scope of these changes?

File: pkg/apis/sessions/session_state.go [hunk 2]
```diff
@@ -164,12 +164,11 @@ func into(s *string, f codecFunc) error {
 // algorithms.
 func lz4Compress(payload []byte) ([]byte, error) {
 	buf := new(bytes.Buffer)
-	zw := lz4.NewWriter(nil)
-	zw.Header = lz4.Header{
-		BlockMaxSize:     65536,
-		CompressionLevel: 0,
-	}
-	zw.Reset(buf)
+	zw := lz4.NewWriter(buf)
```
Comment (NickMeves) [inline pkg/apis/sessions/session_state.go:167]: Related to the Redis comment -- is this required by the 1.15 Go bump? Or is this better served as its own PR to keep the scope of changes separate?

Do you have a link to the v4 library for these LZ4 API changes? I'm struggling to find it.

File: providers/logingov.go [hunk 3]
```diff
@@ -35,7 +36,11 @@ var letters = []rune("abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ")
 func randSeq(n int) string {
 	b := make([]rune, n)
 	for i := range b {
-		b[i] = letters[rand.Intn(len(letters))]
+		n, err := rand.Int(rand.Reader, big.NewInt(int64(len(letters))))
+		if err != nil {
+			panic(err)
```
Comment (NickMeves) [inline providers/logingov.go:41]: Let's return an error instead of `panic`ing here.

In the `NewLoginGovProvider` usage of this, I think it is then OK to panic if this goes bad (since this is during startup and the `New` interface for providers that consumes this doesn't do errors).

But in `Redeem` usage of this, we want to bubble up an error appropriately so the program doesn't crash but an error 500 is appropriately returned to the end user.

File: providers/logingov.go [hunk 4]
```diff
@@ -3,10 +3,11 @@ package providers
 import (
 	"bytes"
 	"context"
+	"crypto/rand"
```
Comment (NickMeves) [inline providers/logingov.go:6]: Thanks for taking on this change -- good security improvement! 👍

File: pkg/validation/options.go [hunk 5]
```diff
@@ -41,7 +41,9 @@ func Validate(o *options.Options) error {
 		if err == nil {
 			transport := &http.Transport{
 				TLSClientConfig: &tls.Config{
-					RootCAs: pool,
+					RootCAs:    pool,
+					MinVersion: tls.VersionTLS12,
```
Comment (NickMeves) [inline pkg/validation/options.go:45]: Let's split this change into its own PR.  This is definitely a breaking change & warrants its own entry in the CHANGELOG.

I'll need to review what the EOL schedule of popular browsers is for TLS 1.0 & 1.1 (in particular browsers that are in the corporate/enterprise space since they seem to lag on browser updates compared to individuals and they are our userbase).

**Comment by NickMeves** [comment]
@johejo Can we leave this PR open to get the Go update portions of it?

I think all these changes are very valid & needed -- lets just do some of the areas I mentioned in separate PRs so we can get good documentation of the changes in the CHANGELOG.

Feel free to re-close if you disagree.

**Comment by johejo** [comment]
Thanks good feedbacks.
I will split PR.
